### PR TITLE
Add stacktrace-translate tool

### PR DESCRIPTION
**Pull request by coderkevin**
This adds a tool to translate a minified stacktrace to a meaningful one.

To Test:

1. Find an error from logstash in kibana that you want to investigate.
2. Copy the commit sha from the "commit" field.
3. Run `git checkout <commit sha>`
4. Run `CALYPSO_ENV=production SOURCEMAP=hidden-source-map npm run build`
5. Run `git checkout add/stacktrace-translate-tool` (only necessary until this PR is merged)
6. Run `npm install` (to ensure the extra npm dependency is present)
7. Going back to kibana, copy the "trace" field to your clipboard. (I use triple-click to select)
8. Run `./bin/stacktrace-translate.js -v`
9. Paste the trace into the terminal.
10. Observe the results.

I've checked the results for several errors, and while some do work exactly as expected, others do not yet because of minified `node_module`bundles that are re-bundled into our own. Also, the traces seem to be one stack frame below the actual error, so if you search for error text, you likely won't find it in stack frame [0] code.

Further work may be needed to perhaps improve the stack traces given to logstash, and to un-minify dependent modules (if that's what we really want to do).

### CODE REVIEW

**Comment by timmyc** [comment]
I gave it a try using the steps above but it didn't seem to work for me. Let's work on it IRL in a few days 🌞

**Comment by sirreal** [comment]
Link to relevant Kibana query: 1d061-pb

**Comment by sirreal** [comment]
We can cherry pick the required changes directly onto the commit which can simplify the process (4 -7)

> 4. Before running the next step, you will have to modify `package.json` in accordance with PR #22066 or you will likely run out of memory. (only necessary until that #22066 gets merged)
> 5. Run `CALYPSO_ENV=production SOURCEMAP=hidden-source-map npm run build`
> 6. Run `git checkout -- package.json` so you can do another checkout (only necessary until #22066 is merged)
> 7. Run `git checkout add/stacktrace-translate-tool` (only necessary until this PR is merged)
> 

Become:

```sh
git cherry-pick --no-commit e80537e59977543f93e6b3b12c83faf7b31e04b8 05896b8aa9bc29e4aeacef418722acb2d2b874e5
```

Just remember there are changes to your working tree now 😉

**Comment by coderkevin** [comment]
I've tested out this PR again, and it appears to be working for chrome error logs. However, it may not be working for errors logged by other browsers. If errors aren't being parsed correctly, please post information about the browser and type from the logstash info.

**Comment by coderkevin** [comment]
Thanks for testing @sirreal.

> Here are a sample of some errors. These seem to be mal-formed error input:

I'm guessing you grabbed a non-Chrome error. I think maybe browsers aren't outputting the same format for error stack traces, and we may need to standardize that via stacktrace.js. So I think this may only work for Chrome errors right now.

The following concerns me a bit more:

That's because the library used to decode the stacktrace is designed to pull .map files from URLs, but this code runs it in offline mode and provides the .map files locally. You're seeing failed requests for vendor bundles there. It's not harmful, but perhaps could be improved.

**Comment by timmyc** [comment]
@sirreal any chance you could give this one more glance? It might be nice to get this merged in and iterate further - but I definitely feel that as-is this tool could be used by others in Calypso to track down and fix 🐛

**Comment by sirreal** [comment]
@timmyc I'm really swamped right now, but I won't stand in the way if other folks want to review and move ahead with this 🙂 

I've dismissed my previous review so it shouldn't block any progress here.

**Comment by justinshreve** [comment]
@coderkevin @timmyc Who would be a good alternative person to ping for a review here? 

I've used this tool successfully on a couple errors now and it'd be awesome to have it available without having to do the branch switching, even if the errors are still limited to chrome. I don't feel like I'm the best person to sign off on it, though nothing is jumping out at me.

**Comment by coderkevin** [comment]
rebased on current master

File: bin/stacktrace-translate.js [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+	const fileNames = glob.sync( SOURCE_PATTERN );
+	const files = {};
+
+	fileNames.forEach( ( fileName ) => {
+		const relativeFileName = fileName.substring( filePrefixChars );
+		const url = BUNDLE_URL_ROOT + relativeFileName;
+		const text = fs.readFileSync( fileName );
+		const textWithMapping = text + '\n//# sourceMappingURL=' + url + '.map';
+		files[ url ] = textWithMapping;
+	} );
```
[the remark this answers is not in the document]
Comment (coderkevin) [reply bin/stacktrace-translate.js]: Yeah, I may try that to see if it increases performance.

**Pull request merged**